src/tsserver/nodeServer.ts

```diff
@@ -779,10 +779,21 @@ namespace ts.server {
                 return JSON.stringify(message, undefined, 2);
             }
 
+            public exit() {
+                this.logger.info("Exiting...");
+                this.projectService.closeLog();
+                tracing?.stopTracing();
+                process.exit(0);
+            }

```

Comment:
```suggestion
```
Given ` IpcIOSession extends IOSession` we dont need to duplicate this. I think the missing piece was onDisconnect which you have added. Can you pls check if things are working with just that. Thanks

Reply:
That's perfect, I've totally missed this fact.
Yes, fix works after removing the override.